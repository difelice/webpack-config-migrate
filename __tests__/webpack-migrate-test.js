import webpackMigrate from "../index";
import webpack from "webpack";

describe("webpack migrate", () => {
  it("matches v1 to v2 snapshot", () => {
    expect(webpackMigrate([V1_CONFIG])).toMatchSnapshot();
  });
});

const V1_CONFIG = {
  context: `${__dirname}/src`,
  entry: {
    admin_app: "./admin/app.js"
  },
  output: {
    path: "../static/dist",
    filename: "[name].js"
  },
  module: {
    loaders: [
      {
        test: /\.js?$/,
        exclude: [/node_modules/, /vendor/],
        loader: "babel"
      },
      { test: /\.less$/, loader: "style-loader!css-loader!less-loader" },
      { test: /\.css$/, loader: "style-loader!css-loader" },
      { test: /\.json$/, loader: "json-loader" }
    ]
  },
  plugins: [
    new webpack.optimize.DedupePlugin(),
    new webpack.optimize.OccurrenceOrderPlugin()
  ],
  externals: {
    jquery: "jQuery"
  },
  resolveLoader: {
    root: "./node_modules"
  },
  resolve: {
    root: ["./app_modules"]
  }
};
