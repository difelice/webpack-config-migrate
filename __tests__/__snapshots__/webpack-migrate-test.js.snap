// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`webpack migrate matches v1 to v2 snapshot 1`] = `
Array [
  Object {
    "context": "/Users/admin/git/webpack-config-migrate/__tests__/src",
    "entry": Object {
      "admin_app": "./admin/app.js",
    },
    "externals": Object {
      "jquery": "jQuery",
    },
    "module": Object {
      "rules": Array [
        Object {
          "exclude": Array [
            /node_modules/,
            /vendor/,
          ],
          "test": /\\\\\\.js\\?\\$/,
          "use": Array [
            "babel",
          ],
        },
        Object {
          "test": /\\\\\\.less\\$/,
          "use": Array [
            "style-loader",
            "css-loader",
            "less-loader",
          ],
        },
        Object {
          "test": /\\\\\\.css\\$/,
          "use": Array [
            "style-loader",
            "css-loader",
          ],
        },
      ],
    },
    "output": Object {
      "filename": "[name].js",
      "path": "../static/dist",
    },
    "plugins": Array [],
    "resolve": Object {
      "modules": Array [
        "./app_modules",
      ],
    },
    "resolveLoader": Object {},
  },
]
`;
